### ISSUE TITLE
METRLA点缺失实验结果

### ISSUE THREAD

**Issue by yw666758-boop** [JeremyChou28/MTSCI#9]
作者您好，
        我在复现论文的时候发现METRLA点缺失实验结果MAE只能达到1.859左右和论文一直有些差距，我使用了代码中的默认配置，但一直没有达到论文中的结果。是不是需要调整一下缺失率或者超参数等，有没有什么办法。
非常期待您的回复，十分感谢。

**Comment by JeremyChou28** [comment]
我找了下我当时一个日志文件，你可以参考下~


Namespace(checkpoint_path='../saved_models/METR-LA/point/0.2/model_2024-05-22-20-09-19.pth', dataset='METR-LA', dataset_path='../datasets/METR-LA/', device='cuda:4', feature=207, missing_pattern='point', missing_ratio=0.2, nsample=100, save_result_path='../results/', saving_path='../saved_models', scratch=False, seed=1, seq_len=24, test_missing_ratio=0.2, val_missing_ratio=0.2)
2024-05-23-01-05-53
{
    "train": {
        "epochs": 300,
        "batch_size": 4,
        "lr": 0.001,
        "alpha": 1,
        "beta": 10
    },
    "diffusion": {
        "seqlen": 24,
        "layers": 4,
        "channels": 64,
        "nheads": 8,
        "diffusion_embedding_dim": 128,
        "beta_start": 0.0001,
        "beta_end": 0.2,
        "num_steps": 50,
        "schedule": "quad"
    },
    "model": {
        "timeemb": 128,
        "featureemb": 16
    }
}
train data shape:  (23990, 207)
Train: original missing ratio = 0.0713, artificial missing ratio = 0.2003, artificial missing pattern: point, overall missing ratio = 0.2573
train samples: 998
val data shape:  (3427, 207)
val: original missing ratio = 0.0696, artificial missing ratio = 0.1998, artificial missing pattern: point, overall missing ratio = 0.2554
val samples: 142
test data shape:  (6855, 207)
test: original missing ratio = 0.1210, artificial missing ratio = 0.2001, artificial missing pattern: point, overall missing ratio = 0.2969
test samples: 285
len train dataloader:  250
len val dataloader:  36
len test dataloader:  72
load model from ../saved_models/METR-LA/point/0.2/model_2024-05-22-20-09-19.pth

  0%|          | 0/72 [00:00<?, ?it/s]
  1%|▏         | 1/72 [00:40<48:03, 40.61s/it]
  3%|▎         | 2/72 [01:23<48:49, 41.85s/it]
  4%|▍         | 3/72 [02:04<47:51, 41.62s/it]
  6%|▌         | 4/72 [02:47<47:54, 42.27s/it]
  7%|▋         | 5/72 [03:27<46:05, 41.28s/it]
  8%|▊         | 6/72 [04:07<44:45, 40.69s/it]
 10%|▉         | 7/72 [04:46<43:45, 40.40s/it]
 11%|█         | 8/72 [05:26<42:49, 40.15s/it]
 12%|█▎        | 9/72 [06:08<42:55, 40.88s/it]
 14%|█▍        | 10/72 [06:48<41:49, 40.47s/it]
 15%|█▌        | 11/72 [07:29<41:09, 40.49s/it]
 17%|█▋        | 12/72 [08:10<40:56, 40.95s/it]
 18%|█▊        | 13/72 [08:50<39:51, 40.54s/it]
 19%|█▉        | 14/72 [09:30<38:54, 40.25s/it]
 21%|██        | 15/72 [10:09<38:04, 40.07s/it]
 22%|██▏       | 16/72 [10:49<37:16, 39.94s/it]
 24%|██▎       | 17/72 [11:28<36:28, 39.79s/it]
 25%|██▌       | 18/72 [12:11<36:38, 40.71s/it]
 26%|██▋       | 19/72 [12:51<35:39, 40.37s/it]
 28%|██▊       | 20/72 [13:30<34:45, 40.11s/it]
 29%|██▉       | 21/72 [14:10<33:59, 39.98s/it]
 31%|███       | 22/72 [14:50<33:13, 39.88s/it]
 32%|███▏      | 23/72 [15:29<32:31, 39.83s/it]
 33%|███▎      | 24/72 [16:09<31:48, 39.77s/it]
 35%|███▍      | 25/72 [16:53<32:05, 40.97s/it]
 36%|███▌      | 26/72 [17:32<31:04, 40.53s/it]
 38%|███▊      | 27/72 [18:12<30:10, 40.23s/it]
 39%|███▉      | 28/72 [18:51<29:19, 39.99s/it]
 40%|████      | 29/72 [19:31<28:32, 39.83s/it]
 42%|████▏     | 30/72 [20:10<27:47, 39.71s/it]
 43%|████▎     | 31/72 [20:50<27:04, 39.62s/it]
 44%|████▍     | 32/72 [21:29<26:24, 39.60s/it]
 46%|████▌     | 33/72 [22:09<25:43, 39.57s/it]
 47%|████▋     | 34/72 [22:48<25:01, 39.51s/it]
 49%|████▊     | 35/72 [23:27<24:21, 39.50s/it]
 50%|█████     | 36/72 [24:07<23:40, 39.46s/it]
 51%|█████▏    | 37/72 [24:46<23:01, 39.48s/it]
 53%|█████▎    | 38/72 [25:26<22:22, 39.48s/it]
 54%|█████▍    | 39/72 [26:05<21:42, 39.48s/it]
 56%|█████▌    | 40/72 [26:45<21:03, 39.49s/it]
 57%|█████▋    | 41/72 [27:24<20:24, 39.49s/it]
 58%|█████▊    | 42/72 [28:04<19:44, 39.48s/it]
 60%|█████▉    | 43/72 [28:43<19:04, 39.48s/it]
 61%|██████    | 44/72 [29:23<18:25, 39.48s/it]
 62%|██████▎   | 45/72 [30:02<17:45, 39.48s/it]
 64%|██████▍   | 46/72 [30:42<17:06, 39.49s/it]
 65%|██████▌   | 47/72 [31:21<16:27, 39.50s/it]
 67%|██████▋   | 48/72 [32:01<15:47, 39.49s/it]
 68%|██████▊   | 49/72 [32:40<15:07, 39.46s/it]
 69%|██████▉   | 50/72 [33:19<14:27, 39.44s/it]
 71%|███████   | 51/72 [33:59<13:48, 39.44s/it]
 72%|███████▏  | 52/72 [34:38<13:08, 39.42s/it]
 74%|███████▎  | 53/72 [35:18<12:29, 39.45s/it]
 75%|███████▌  | 54/72 [35:57<11:50, 39.47s/it]
 76%|███████▋  | 55/72 [36:37<11:11, 39.48s/it]
 78%|███████▊  | 56/72 [37:16<10:31, 39.49s/it]
 79%|███████▉  | 57/72 [37:56<09:52, 39.49s/it]
 81%|████████  | 58/72 [38:35<09:12, 39.48s/it]
 82%|████████▏ | 59/72 [39:15<08:33, 39.48s/it]
 83%|████████▎ | 60/72 [39:54<07:53, 39.48s/it]
 85%|████████▍ | 61/72 [40:34<07:14, 39.49s/it]
 86%|████████▌ | 62/72 [41:13<06:35, 39.51s/it]
 88%|████████▊ | 63/72 [41:53<05:55, 39.51s/it]
 89%|████████▉ | 64/72 [42:32<05:15, 39.49s/it]
 90%|█████████ | 65/72 [43:12<04:36, 39.47s/it]
 92%|█████████▏| 66/72 [43:51<03:56, 39.45s/it]
 93%|█████████▎| 67/72 [44:31<03:17, 39.44s/it]
 94%|█████████▍| 68/72 [45:10<02:37, 39.42s/it]
 96%|█████████▌| 69/72 [45:49<01:58, 39.44s/it]
 97%|█████████▋| 70/72 [46:29<01:19, 39.52s/it]
 99%|█████████▊| 71/72 [47:09<00:39, 39.51s/it]
100%|██████████| 72/72 [47:52<00:00, 40.72s/it]
100%|██████████| 72/72 [47:52<00:00, 39.90s/it]
mae = 1.660, rmse = 3.086, mape = 2.898%, mse = 9.525, r2 = 0.944, CRPS = 0.0221
Spend Time:  2877.2397894859314
